jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java

```diff
@@ -120,6 +124,12 @@ public Response registerInstallation(
         // otherwise we register a new installation:
         logger.finest("Mobile Application on device was launched");
 
+        //let's do update the analytics
+
+        String pushIdentifier = extractPushIdentifier(request);
+        if(pushIdentifier != null) {
```

Comment:
push identifier is the key concept of this PR ;) but let me document the extratPushIdentifier method.